File: tcsh/ruby/test-front-end.rb
#! /usr/bin/env ruby

if ENV.has_key? "TCSH_RUBY_PATH"
  $:.concat ENV["TCSH_RUBY_PATH"].split(":")
end

if ARGV.size != 1
  puts "Usage: ti.rb file.tig"
  exit 1
end

require "tempfile"
require "tc"


# Parse the file.
lib = Tc::Misc::File_library.new()
ast = Tc::Parse::parse("builtin", ARGV[0], lib)

if Tc::has? "Bind" then
  Tc::Bind::bind ast
end

if Tc::has? "Type" then
  Tc::Type::types_check ast
end

if Tc::has? "Translate" then
  # Compute HIR.
  fragments = Tc::Translate::translate ast

  # Dump it.
  hir = Tempfile.new "hir"
  hir.print fragments
  hir.flush

  # Run havm on it.
  system("havm", hir.path)
  hir.close
end

File: tcsh/python/test-front-end.py
#! /usr/bin/env python

import sys
import tc
import os
import tempfile

if len (sys.argv) != 2:
  print "Usage: ti.py file.tig"
  sys.exit (1)

# FIXME: Implement error handling.  Currently, we don't check the
# result of any stage!

# Parse the file.
lib = tc.misc.file_library ()
ast = tc.parse.parse ("builtin", sys.argv[1], lib)

if tc.has ("bind"):
  tc.bind.bind (ast)

if tc.has ("type"):
  tc.type.types_check (ast)

if tc.has ("translate"):
  # Compute HIR.
  fragments = tc.translate.translate (ast)

  # Dump it.
  hir = tempfile.NamedTemporaryFile ()
  print >> hir, fragments
  hir.flush ()

  # Run havm on it.
  os.system ("havm " + hir.name)

# Local Variables:
# mode: python
# End:
